automation/release.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+set -exuo pipefail
+
+function cleanup_gh_install() {
+    rm -f "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    rm -rf "gh_${GH_CLI_VERSION}_linux_amd64/"
+}
+
+function ensure_gh_cli_installed() {
+    if command -V gh; then
+        return
+    fi
+
+    trap 'cleanup_gh_install' EXIT SIGINT SIGTERM
+
+    # install gh cli for uploading release artifacts, with prompt disabled to enforce non-interactive mode
+    curl -sSL "https://github.com/cli/cli/releases/download/v${GH_CLI_VERSION}/gh_${GH_CLI_VERSION}_linux_amd64.tar.gz" -o "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    tar xvf "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    sudo cp "gh_${GH_CLI_VERSION}_linux_amd64/bin/gh" /usr/local/bin/
+    if ! command -V gh; then
+        echo "gh cli not installed successfully"
+        exit 1
+    fi
+    gh config set prompt disabled
+}
+
+function main() {
+
+    DOCKER_TAG="$(git tag --points-at HEAD | head -1)"
+    if [ -z "$DOCKER_TAG" ]; then
+        exit 0
```

Comment:
@dhiller should we produce some log to the stdout in case we unable to detect a tag?

Reply:
I don't think so, because we only want to handle the case when we are on a tag. I figure you mean something like an echo that tells us we are not on a tag?

Reply:
@dhiller Oh I understand, thank you for the clarification. It's actually a sane scenario when this script is being called and a tag doesn't exist.